site/blog/unlisted/save-on-ecr.mdx

```diff
@@ -0,0 +1,37 @@
+---
+slug: save-on-ecr
+title: Save on AWS by deleting untagged ECR images
+authors: [depombo]
+---
+
+# Save on AWS by deleting untagged ECR images used in ECS or EKS
+
+Every time you want to deploy a new version of your code to an ECS container or EKS pod, you build and push a new docker image from your code into an ECR container repository. The latest image in an ECR repository is the only one in use. Stale images remain in the repository and accumulate over long periods by normal usage of ECS, or EKS, as there is typically no workflow to delete untagged images. However, there is a gotcha. AWS charges $0.10 per GB per month for images stored in private or public ECR repositories. This doesn't sound like a lot, but for context, the size of the IaSQL docker image is around 600 MB. We have a CI job that deploys our staging environment every time we land a pull request to the main branch of our repository. This added up to 845 deploys over only a few months. Those 845 images, which have since been deleted, summed up to roughly 50 GB which was $50 per month in our case. However, companies with lots of microservices on ECS or large docker containers can have many gigabytes of unused storage that can come out to hundreds or thousands of dollars per month of unnecessary AWS spend.
```

Comment:
Hmm...
> AWS charges $0.10 per GB per month for images stored ...

and

> summed up to roughly 50 GB

I expected it to be $5, but the text says:

> $50 per month in our case

I think I'm missing a point here.

Reply:
good catch, I need to fix the math